sync2/poller.go

```diff
@@ -288,7 +289,7 @@ func (h *PollerMap) EnsurePolling(pid PollerID, accessToken, v2since string, isS
 
 	// replace the poller. If we don't need to wait, then we just want to nab to-device events initially.
 	// We don't do that on startup though as we cannot be sure that other pollers will not be using expired tokens.
-	poller = newPoller(pid, accessToken, h.v2Client, h, logger, !needToWait && !isStartup)
```

Comment:
This `logger` has contextual information like the user ID and device ID which is now lost.

Reply:
This was a conscious decision. In the implementation of `newPoller()` there are calls like this:
```
logger.Error().Str("user", p.userID).Str("device", p.deviceID).Msgf("%s. Traceback:\n%s", panicErr, debug.Stack())
```
where context is sent to the log output even though the logger already has that context. My worry is this happened because it wasn't clear that the logger already has that context or because it wasn't added in an earlier (and now-deleted) path or just to ensure the context is there in case one day it's handed a logger without it.

In any case, it makes more sense to me to add those values locally. There are several other logging calls in `newPoller()` that don't explicitly add user/device to output but it doesn't seem particularly useful to have it at those points. If that's not the case, my inclination would be to add those values locally instead of far up the call stack.

Going back to using the per-file local loggers would mean plumbing each of the dozen or so `zerolog.New()` call sites with the new environment variable logic.

I've rebased the PR on the current main branch to accommodate new code changes and removed the unused logger formal parameter. Please let me know if this works for you.

Reply:
I would rather we double reported the user ID and device ID than risked not reporting the user ID / device ID at all. By having a poller-local `logger` instance we guarantee that those keys are always logged. We rely on these keys existing for debugging as we often search for these IDs when that user reports a problem.

I'm happy with the global logger usage everywhere else, but the poller really should be using the contextual one please.